modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/prototype/PrototypedAPITestcase.java

```diff
@@ -0,0 +1,305 @@
+        apiRequest.setProvider(apiProvider);
+
+        HttpResponse addAPIResponse = restAPIPublisher.addAPI(apiRequest);
+        assertEquals(addAPIResponse.getResponseCode(), Response.Status.CREATED.getStatusCode(),
+                "Invalid Response Code");
+        apiID = addAPIResponse.getData();
+
+        //Deployed API as a Prototyped API & check the status
+        WorkflowResponseDTO lcChangeResponse = restAPIPublisher.changeAPILifeCycleStatus(apiID,
+                APILifeCycleAction.DEPLOY_AS_PROTOTYPE.getAction());
+
+        HttpResponse response = restAPIPublisher.getAPI(apiID);
+        Gson g = new Gson();
+        APIDTO apidto = g.fromJson(response.getData(), APIDTO.class);
+        String endPointString = "{\"implementation_status\":\"prototyped\",\"endpoint_type\":\"http\"," +
+                "\"production_endpoints\":{\"config\":null," +
+                "\"url\":\"" + apiEndPointUrl + "\"}," +
+                "\"sandbox_endpoints\":{\"config\":null,\"url\":\"" + apiEndPointUrl + "\"}}";
+
+        JSONParser parser = new JSONParser();
+        JSONObject endpoint = (JSONObject) parser.parse(endPointString);
+        apidto.setEndpointConfig(endpoint);
+
+        restAPIPublisher.updateAPI(apidto);
+
+        assertTrue(lcChangeResponse.getLifecycleState().getState().equals("Prototyped"),
+                apiName + "  status not updated as Prototyped");
+        waitForAPIDeployment();
+
+        // Create a revision and Deploy the API
+        createAPIRevisionAndDeployUsingRest(apiID, restAPIPublisher);
+
+        //Check whether Prototype API is available in publisher
+        APIListDTO getAllAPIsResponse = restAPIPublisher.getAllAPIs();
+        assertTrue(APIMTestCaseUtils.isAPIAvailable(apiIdentifier, getAllAPIsResponse),
+                "Implemented" + apiName + " Api is not visible in API Publisher.");
+        waitForAPIDeployment();
+
+        //Check whether Prototype API is available under the Prototyped API
+        org.wso2.am.integration.clients.store.api.v1.dto.APIListDTO prototypedAPIs = restAPIStore
+                .getPrototypedAPIs(user.getUserDomain());
+        assertTrue(APIMTestCaseUtils.isAPIAvailableInStore(apiIdentifier, prototypedAPIs),
+                apiName + " is not visible as Prototyped API");
+
+        Map<String, String> requestHeaders = new HashMap<>();
+        requestHeaders.put("accept", "application/json");
+
+        //Invoke the Prototype endpoint and validate
+        HttpResponse response1 = HTTPSClientUtils
+                .doGet(getAPIInvocationURLHttps(apiContext, apiVersion) +
+                        "", requestHeaders);
+        Assert.assertEquals(response1.getResponseCode(), 200);
+        Assert.assertTrue(response1.getData().contains("BBQ Chicken Bacon"));
+
+        restAPIPublisher.changeAPILifeCycleStatus(apiID, APILifeCycleAction.DEMOTE_TO_CREATE.getAction());
+        restAPIPublisher.deleteAPI(apiID);
```

Comment:
We don't need to make the API created state when deleting.
Also, its better to get all deletions into one  @AfterClass(alwaysRun = true) method.. Otherwise, if a certain assertion fails, this line will not be executed and the tests will not delete the API.

Reply:
Fixed